libs/corda-sdk/src/main/kotlin/net/corda/sdk/network/RegistrationRequester.kt

```
@@ -0,0 +1,92 @@
+
+    fun requestRegistration(
+        restClient: RestClient<MemberRegistrationRestResource>,
+        registrationContext: Map<String, Any?>,
+        holdingId: String,
+        wait: Duration = 10.seconds
+    ): RegistrationRequestProgress {
+        val castRegistrationContext: Map<String, String> = registrationContext.mapValues { (_, value) ->
+            value.toString()
+        }
+        val request = MemberRegistrationRequest(
+            context = castRegistrationContext,
+        )
+        return restClient.use { client ->
+            executeWithRetry(
+                waitDuration = wait,
+                operationName = "Request registration"
+            ) {
+                val resource = client.start().proxy
+                resource.startRegistration(holdingId, request)
+            }
+        }
+    }
+
+    fun waitForRegistrationApproval(
+        restClient: RestClient<MemberRegistrationRestResource>,
+        registrationId: String,
+        holdingId: String,
+        wait: Duration = 10.seconds
+    ) {
+        restClient.use { client ->
+            executeWithRetry(
+                waitDuration = wait,
+                operationName = "Check registration progress"
+            ) {
+                val resource = client.start().proxy
+                val status = resource.checkSpecificRegistrationProgress(holdingId, registrationId)
+                when (val registrationStatus = status.registrationStatus) {
+                    RegistrationStatus.APPROVED -> true // Return true to indicate the condition is satisfied
+                    RegistrationStatus.DECLINED,
+                    RegistrationStatus.INVALID,
+                    RegistrationStatus.FAILED,
+                    -> throw OnboardException("Status of registration is $registrationStatus; reason: ${status.reason}.")
+                    else -> throw OnboardException("Status of registration is $registrationStatus; reason: ${status.reason}.")
```

Are you sure about that? Should we throw an exception if the registration is still ongoing?

---

The exception is caught by the retry, and will continue to loop until timing out